test/exercise/fp2/solution.rb

```diff
@@ -1,23 +1,37 @@
 module Exercise
   module Fp2
     class MyArray < Array
-      # Использовать стандартные функции массива для решения задач нельзя.
-      # Использовать свои написанные функции для реализации следующих - можно.
-
-      # Написать свою функцию my_each
       def my_each
+        for element in self
+          yield element
+        end
       end
 
-      # Написать свою функцию my_map
       def my_map
+        new_array = MyArray.new
```

Comment:
Try to implement `my_map` and `my_compact` using reduce.

Reply:
done